cli/src/rpc.rs

```diff
@@ -276,7 +280,9 @@ impl<S: Serialization, C: Send + Sync + 'static> RpcMethodBuilder<S, C> {
 		self.register_async(METHOD_STREAM_ENDED, move |m: StreamEndedParams, _| {
 			let s1 = s1.clone();
 			async move {
-				s1.lock().await.remove(&m.stream);
+				if let Some(mut s) = s1.lock().await.remove(&m.stream) {
+					let _ = s.shutdown().await;
```

Comment:
Previously, this didn't actually "end" the stream, and now we want to be able to close stdin to signal nested clis to shut down